src/runtime/hexagon/fastrpc/tvm_hexagon_remote.h

```diff
@@ -0,0 +1,138 @@
+ *      _modver: module version, _modver=1.0
+ *      _*: any other key name starting with an _ is reserved
+ *    Unknown uri keys/values are forwarded as is.
+ * @param h, resulting handle
+ * @retval, 0 on success
+ */
+__QAIC_HEADER_EXPORT int __QAIC_HEADER(tvm_hexagon_remote_open)(
+    const char* uri, remote_handle64* h) __QAIC_HEADER_ATTRIBUTE;
+/**
+    * Closes a handle.  If this is the last handle to close, the session
+    * is closed as well, releasing all the allocated resources.
+
+    * @param h, the handle to close
+    * @retval, 0 on success, should always succeed
+    */
+__QAIC_HEADER_EXPORT int __QAIC_HEADER(tvm_hexagon_remote_close)(
+    remote_handle64 h) __QAIC_HEADER_ATTRIBUTE;
+typedef struct _tvm_hexagon_remote_buffer__seq_octet
+    _tvm_hexagon_remote_buffer__seq_octet;
+typedef _tvm_hexagon_remote_buffer__seq_octet tvm_hexagon_remote_buffer;
+struct _tvm_hexagon_remote_buffer__seq_octet {
+  unsigned char* data;
+  int dataLen;
+};
+typedef unsigned int tvm_hexagon_remote_handle_t;
+typedef uint64 tvm_hexagon_remote_scalar_t;
+__QAIC_HEADER_EXPORT int __QAIC_HEADER(tvm_hexagon_remote_map_buffer_to_dsp)(
+    remote_handle64 _h, int buffer_fd, unsigned int buffer_fd_len,
+    uint64* dsp_va) __QAIC_HEADER_ATTRIBUTE;
+__QAIC_HEADER_EXPORT int __QAIC_HEADER(tvm_hexagon_remote_unmap_buffer_to_dsp)(
+    remote_handle64 _h, uint64 dsp_va,
+    tvm_hexagon_remote_handle_t bufferLen) __QAIC_HEADER_ATTRIBUTE;
+__QAIC_HEADER_EXPORT int __QAIC_HEADER(tvm_hexagon_remote_load_library)(
+    remote_handle64 _h, const char* soname, int sonameLen,
+    const char* code, int codeLen,
+    tvm_hexagon_remote_handle_t* module_ptr) __QAIC_HEADER_ATTRIBUTE;
+__QAIC_HEADER_EXPORT int __QAIC_HEADER(tvm_hexagon_remote_get_symbol)(
+    remote_handle64 _h, tvm_hexagon_remote_handle_t module_ptr,
+    const char* name, int nameLen,
+    tvm_hexagon_remote_handle_t* sym_ptr) __QAIC_HEADER_ATTRIBUTE;
+__QAIC_HEADER_EXPORT int __QAIC_HEADER(tvm_hexagon_remote_kernel)(
+    remote_handle64 _h, tvm_hexagon_remote_handle_t module_ptr,
+    tvm_hexagon_remote_handle_t symbol, int* scalar, int scalarLen, int* stack,
+    int stackLen, const tvm_hexagon_remote_buffer* scalar_in_octet,
+    int scalar_in_octetLen, tvm_hexagon_remote_buffer* scalar_out_octet,
+    int scalar_out_octetLen, const tvm_hexagon_remote_buffer* stack_in_octet,
+    int stack_in_octetLen, tvm_hexagon_remote_buffer* stack_out_octet,
+    int stack_out_octetLen, uint64* pcycles,
+    uint64* time_usec) __QAIC_HEADER_ATTRIBUTE;
+__QAIC_HEADER_EXPORT int __QAIC_HEADER(tvm_hexagon_remote_release_library)(
+    remote_handle64 _h,
+    tvm_hexagon_remote_handle_t module_ptr) __QAIC_HEADER_ATTRIBUTE;
+#ifndef tvm_hexagon_remote_URI
+#define tvm_hexagon_remote_URI                                            \
+  "file:///"                                                              \
+  "libtvm_hexagon_remote_skel.so?tvm_hexagon_remote_skel_handle_invoke&_" \
```

Comment:
`libtvm_hexagon_remote_skel.so` is not in HexagonSDK, we build it. Could we also upload it?

Reply:
I'll upload the sources next week.  You could use them to build the skel.so and the stub.so.

Reply:
There is a bit of delay.  Hopefully I'll get that done next week.